chamber/models/__init__.py

```diff
@@ -202,28 +203,75 @@ def _persistence_clean(self):
             else:
                 raise PersistenceException(', '.join(map(force_text, er.messages)))
 
-    def save(self, clean_before_save=None, clean_after_save=None, clean_before_delete=None, clean_after_delete=None,
-             force_insert=False, force_update=False, using=None, update_fields=None, *args, **kwargs):
-        clean_before_save = self._smart_meta.clean_before_save if clean_before_save is None else clean_before_save
+    def _get_save_extra_kwargs(self):
+        return {}
+
+    def _pre_save(self, *args, **kwargs):
+        pass
+
+    def _save(self, clean_pre_save=None, clean_post_save=None, force_insert=False, force_update=False, using=None,
```

Comment:
we should now rewrite only method with _ prefix because without prefix is performing atomicity